packages/@azure/servicebus/data-plane/lib/util/utils.ts

```diff
@@ -174,3 +175,15 @@ export function toBuffer(input: any): Buffer {
   log.utils("[utils.toBuffer] The converted buffer is: %O.", result);
   return result;
 }
+
+/**
+ * Throws InvalidOperationError if the current AMQP connection is closed.
+ * @param context The ConnectionContext associated with the current AMQP connection.
+ */
+export function throwErrorIfConnectionClosed(context: ConnectionContext): void {
+  if (context && context.wasConnectionCloseCalled) {
+    const err = new Error("The underlying AMQP connection is closed.");
+    err.name = "InvalidOperationError";
```

Comment:
I think either we should just throw a regular Error, or else have a separate error type so e.g. `e instanceof InvalidOperationError` works in a catch block...

Reply:
Updated to use regular Error